.github/pull_request_template.md

```diff
@@ -0,0 +1,46 @@
+<!-- Thank you for submitting a pull request! Find more information in our
+development guide at
+https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md
+and contact us on #dev in slack. -->
+## Description
+<!-- Elaborate beyond the title of the PR as necessary to help the reviewers.-->
+
+## Related issues and pull requests
+<!-- List any issue that is fixed or related, such as a documentation PR:
+
+* Fixes #issuenumber
+* Related documentation in #issuenumber
+-->
+
+## Documentation
+<!-- 
+To fill in by maintainer with input from contributor:
+Remove x to confirm that no documentation and/or no release notes entry is
+needed, or provide relevant information.-->
+
+- [x] User facing documentation needed
+- [x] Release notes entry needed
```

Comment:
This section is pretty confusing to me.  I believe the goal here is to have a "documentation todo" list, but both are default to checked.
I suggest we rename this section to "Documentation TODOs" or "Review TODOs", so we can add other items for reviewers to check off.  Also, I think they should start unchecked, and be reworded to sound like TODOs.  For example:
```
- [ ] Assess need for user facing documentation
- [ ] Add necessary release notes
```

Reply:
Ideally we get the following info:

* Release notes entry needed? either confirm it is not needed or proposal for text
* Docs affected and needed and then info on if docs is in PR or in separate PR or there is at least a issue to get the docs done

Any ideas how we get that into the template better?

Reply:
For the release notes, the maintainer would edit the description and add them inline.  For the docs, what do you think the maintainer would do?  At the very least, I'd like to see a section where the maintainer says, "we're going to need docs to cover x and y", but ideally there would be changes to the docs in the PR.

Reply:
What about instead of check-boxes we (maintainers) start using labels `docs-needed` and `release-notes-needed` to mark PRs that need a RN/docs.

It has multiple benefits, it makes it possible to automate the creation of the list of commits + whether RN or docs are needed or not and it makes it easy to search for such PRs.

Reply:
If we somehow make that consistently happen .. sure.. its just easier for contributors this way since they cant assign labels..

Reply:
I rejigged it along the idea of @dain to have a kind of questionnaire to answer .